cvat/apps/engine/serializers.py

```diff
@@ -84,12 +85,19 @@ def update_instance(validated_data, parent_instance):
         else:
             instance['task'] = parent_instance
             logger = slogger.task[parent_instance.id]
-        (db_label, created) = models.Label.objects.get_or_create(name=validated_data['name'],
-            **instance)
-        if created:
-            logger.info("New {} label was created".format(db_label.name))
+        if not validated_data.get('id') is None:
+            db_label = models.Label.objects.get(id=validated_data['id'],
+                **instance)
+            db_label.name = validated_data.get('name', db_label.name)
         else:
-            logger.info("{} label was updated".format(db_label.name))
+            db_label = models.Label.objects.create(name=validated_data.get('name'), **instance)
+            logger.info("New {} label was created".format(db_label.name))
+        # (db_label, created) = models.Label.objects.get_or_create(name=validated_data['name'],
```

Comment:
@ActiveChooN , do we need the comment? Should we keep logging code?

Reply:
Fixed